### PR TITLE
Add dev mode

### PR DESCRIPTION
**Pull request by tamird**
This makes it _super_ easy to run a single-node in-memory cluster.

`./cockroach start --dev --insecure` and you're off to the races.

### CODE REVIEW

File: cli/start.go [hunk 1]
```diff
@@ -127,12 +137,16 @@ func runStart(cmd *cobra.Command, args []string) {
...
 
-	err = s.Start(false)
-	if err != nil {
+	if err := s.Start(false); err != nil {
 		log.Errorf("cockroach server exited with error: %s", err)
 		return
 	}
 
+	client := makeDBClient()
+	if err := testutils.SetDefaultRangeReplicaNum(client, 1); err != nil {
```
Comment (bdarnell) [inline cli/start.go]: Doesn't this need to be behind `if Context.TransientSingleNode`? 

We shouldn't include `testutils` in the production binary; if we need this it should go in a different package (maybe just `client`?)

**Comment by BramGruneir** [comment]
This is a great idea.

File: cli/start.go (hunk 1 above)
Comment (tamird) [inline cli/start.go]: Done. Added a TODO for the testutils thing.

File: server/context.go [hunk 2]
```diff
@@ -83,6 +83,9 @@ type Context struct {
 	// communicated between hosts on the gossip network.
 	GossipInterval time.Duration
 
+	// // Enables running the node as a single-node in-memory cluster.
+	TransientSingleNode bool
```
Comment (bdarnell) [inline server/context.go]: It's weird that this is in the `server` package; I would have thought that the `cli` package could handle this entirely from the outside (i.e. as soon as the flags are parsed, `if *transientSIngleNode { ctx.GossipBootstrap = "self=" }`)

**Comment by bdarnell** [comment]
LGTM

**Pull request merged**